### PR TITLE
[Feature] Add support for mask-free keypoints conversion

### LINKED ISSUE #35: Refactor: remove keypoint mask and use confidence only
Currently the keypoint mask is often confused with keypoint confidence.

### PR DESCRIPTION
**Pull request by caizhongang**
Resolves #35 .

- [x] Add `generate_mask_from_confidence` in HumanData
- [x] Add `return_mask` switch in `convert_kps` that allow mask to be not returned  

After much discussion, the team agree that:
- Mask is not necessary when keypoints confidence is present
- Mask will be kept to ensure forward compatibility

### CODE REVIEW

File: mmhuman3d/data/data_structures/human_data.py [hunk 1]
```diff
@@ -915,6 +915,56 @@ def __check_value_len__(self, key: Any, val: Any) -> bool:
                 msg=err_msg, logger=self.__class__.logger, level=logging.ERROR)
         return ret_bool
 
+    def generate_mask_from_confidence(self, keys=None) -> None:
+        """Generate mask from keypoints' confidence. Keypoints that have zero
+        confidence in all occurrences will have a zero mask.Note that the last
```
Comment (LazyBusyYang) [inline mmhuman3d/data/data_structures/human_data.py:920]: Missing space in ` a zero mask.Note that`.

File: mmhuman3d/data/data_structures/human_data.py [hunk 2]
```diff
@@ -915,6 +915,56 @@ def __check_value_len__(self, key: Any, val: Any) -> bool:
...
+            raise TypeError(f'`Keys` must be None, str, or list of str, '
+                            f'got{type(keys)}.')
+
+        update_dict = {}
+        for kpt_key in keys:
+            kpt_array = self.get_raw_value(kpt_key)
+            num_joints = kpt_array.shape[-2]
+            # if all conf of a joint are zero, this joint is masked
+            joint_conf = kpt_array[..., -1].reshape(-1, num_joints)
+            mask_array = (joint_conf > 0).astype(np.int).max(axis=0)
```
Comment (LazyBusyYang) [inline mmhuman3d/data/data_structures/human_data.py:962]: For mask, np.uint8 is enough, which is also used in `convert_kps`.
Except that, np.int will cause warning:
`
DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`.
`

**Review by LazyBusyYang — CHANGES_REQUESTED** [review]
After the warning caused by `np.int` is suppressed, everything is good to merge.

File: mmhuman3d/data/data_structures/human_data.py (hunk 2 above)
Comment (caizhongang) [reply mmhuman3d/data/data_structures/human_data.py]: Fixed in https://github.com/open-mmlab/mmhuman3d/pull/142/commits/5d02720a0288b87459bbd0de6800a847c1af5625.

File: mmhuman3d/data/data_structures/human_data.py (hunk 1 above)
Comment (caizhongang) [reply mmhuman3d/data/data_structures/human_data.py]: Fixed in https://github.com/open-mmlab/mmhuman3d/pull/142/commits/5d02720a0288b87459bbd0de6800a847c1af5625.

**Review by LazyBusyYang — APPROVED** [review]
Everything is OK. Good to go.

**Pull request merged**